virtualizarr/xarray.py

```diff
@@ -50,37 +51,88 @@ def open_virtual_dataset(
         Currently can only be ManifestArray, but once VirtualZarrArray is implemented the default should be changed to that.
     """
 
-    # this is the only place we actually always need to use kerchunk directly
-    vds_refs = kerchunk.read_kerchunk_references_from_file(
-        filepath=filepath,
-        filetype=filetype,
-    )
+    if drop_variables is None:
+        drop_variables = []
+
+    if virtual_array_class is not ManifestArray:
+        raise NotImplementedError()
+
+    if filetype == "zarr_v3":
+        # TODO is there a neat way of auto-detecting this?
```

Comment:
This is a bit ugly - I want to automatically distinguish between non-zarr, zarr v2 (to be read using kerchunk) and zarr v3 (to be read using this code). I guess I will just have to search for `.zgroup`/`zarr.json` files explicitly?

Reply:
@norlandrhagen do you have any thoughts on a neat way to handle this?